CHANGELOG.md

```diff
@@ -6,12 +6,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+* Introduced a thread worker pool for parallel downloads
```

Comment:
```suggestion
* Introduce a common thread worker pool for all parallel downloads
```

Reply:
If we have sync threads set to 10 and download threads set to 10, how many threads can be used to download things at the same time? 10? No. 19, because N sync threads can occupy the thread pool while M sync threads can just do small and easy single-threaded downloads and while N+M=10, you can also have N=1,M=9 so 10 large file downloads and 9 small file downloads at the same time. This is not what the user wants though. I think small downloads should also be put in that same download threadpool.

Reply:
Done in 1feddc9cf8802e72fba55bb0b705cced5e9c679d